Improved interfacing with GtkD for GGPlotD.drawToSurface

ljubobratovicrelja opened the pull request:
Version of `GGPlotD.drawToSurface` method is added, that takes the `surface` object of `Surface` type from GtkD package.

Added test proves that original `drawToSurface` method, and freshly added one produce the same result. If you'd like to have any other test added, please let me know.

ljubobratovicrelja committed:
Improved interfacing with GtkD for GGPlotD.drawToSurface

BlackEdder commented on source/ggplotd/ggplotd.d:
```diff
@@ -233,6 +233,30 @@ struct GGPlotD
+            cairo.Surface cairodSurface = new cairo.Surface(cast(cairod_surface_t*)surface.getSurfaceStruct());
+            drawToSurface(cairodSurface, width, height);
+
+            return new gtkdSurface.Surface(cast(gtkd_surface_t*)cairodSurface);
```
I think this can be simplified to

``` D
return surface;
```

BlackEdder commented on source/ggplotd/ggplotd.d:
```diff
@@ -233,6 +233,30 @@ struct GGPlotD
+        import gtkdSurface = cairo.Surface; // cairo surface module in GtkD package.
+
+        /**
+        Draw the plot to a cairo surface.
```
Could you specify here that it draws to gtkd cairo surface?

And maybe for the original drawToSurface specify that it draws to a cairod Surface.

BlackEdder commented:
Looks good to me. Thanks for also adding the test!

Will merge (and tag new release) when you address my minor comments.

ljubobratovicrelja committed:
Revision refactoring for GtkD GGPlotD.drawToSurface

ljubobratovicrelja commented:
Thanks for noticing the return value simplification. Fixed it.

Regarding the comment clarification, are you OK with the change? To be honest, I'm not really sure this addition will tell a new user what the fuss is about with the cairoD and GtkD Surface. I didn't want to add any additional commenting since I'm not quite sure how I would format it, but I think a bit more extensive explanation about this problem should exist in original `drawToSurface` method. Maybe only a note why the original version takes cairoD type, and why the GtkD version exists. What do you think about this?

BlackEdder commented:
You are right that the comments are a bit minimal at the moment. It might be worth it to add a Params: section that states the parameters more clearly/in a straightforward manner. That way it is not needed to specify which one is the "convenience" function and which one is the original function. The user doesn't really need to know if ggplotd uses cairod under the hood, he/she just needs to know both are supported.

rljubobratovic commented:
> You are right that the comments are a bit minimal at the moment. It might be worth it to add a Params: section that states the parameters more clearly/in a straightforward manner. That way it is not needed to specify which one is the "convenience" function and which one is the original function.

OK, will do.

> The user doesn't really need to know if ggplotd uses cairod under the hood, he/she just needs to know both are supported.

I agree. Good point.

ljubobratovicrelja commented:
> You are right that the comments are a bit minimal at the moment. It might be worth it to add a Params: section that states the parameters more clearly/in a straightforward manner. That way it is not needed to specify which one is the "convenience" function and which one is the original function.

OK, will do.

> The user doesn't really need to know if ggplotd uses cairod under the hood, he/she just needs to know both are supported.

I agree. Good point.

ljubobratovicrelja committed:
Added ddoc comments for GGPlotD.drawToSurface

ljubobratovicrelja commented:
I've added ddoc comments for both methods. Also, removed the statement that GtkD version is the _convenience_ method, because as you've said, unnecessarily reveals the implementation details.

OK to merge?

BlackEdder commented:
Thanks for all the work and putting up with my nitpicking :)

BlackEdder commented:
Fixes #21

ljubobratovicrelja commented:
> Thanks for all the work and putting up with my nitpicking :)

No problem! And thanks to you for sharing such library! :)
